### PR TITLE
Fix/888 include missing tests

### PR DESCRIPTION
**Pull request by rquidute**
### What changed
Fixes missing test cases in the TH UI by replacing a fragile, hardcoded module-name allowlist  with a recursive AST-based inheritance resolver that correctly follows local file imports when discovering Python test classes.   

####  list_python_tests_classes.py — Recursive AST inheritance resolver

  - Replaced the hardcoded module-name allowlist in base_test_classes() with a new recursive function
  _is_matter_base_test_class() that walks the AST inheritance chain transitively across local files.
  - The resolver follows from X import Y imports to sibling .py files on disk by converting dotted module paths to filesystem
   paths (e.g. TC_TLSCERT_Base → TC_TLSCERT_Base.py), which is exactly the pattern used by the missing TC-TLSCERT and TC-TLSCLIENT test classes.
  - Handles aliased imports (from X import Y as Z), same-file intermediate base classes, and circular inheritance via a _visiting cycle guard.
  - Retains a matter.testing.* fallback for installed-wheel packages whose source is not available on disk.
  - Exposes MATTER_BASE_TEST_CLASS_NAME as a module-level constant (previously an inlined string literal).
  - get_command_list() now passes search_dir=python_test_file.parent to enable local-file resolution during discovery.

#### test_list_python_tests_classes.py — New unit test file

  - Comprehensive tests covering load_ignore_list, load_include_list, _is_matter_base_test_class, base_test_classes, and
  get_command_list.
  - Covers positive cases (direct inheritance, same-file intermediate base, cross-file local import, aliased import,
  matter.testing fallback), negative cases, and edge cases (cycle guard, syntax errors, attribute-access bases).

#### test_python_test_initialization.py — Test isolation fix

  - Added an autouse fixture restore_singleton_state to save and restore TestScriptManager singleton state around each test,
  preventing async tests that mock discover_test_collections from permanently corrupting test_collections for subsequent
  tests in the same pytest worker (caused 44 unrelated test failures).
  
### Related Issue
https://github.com/project-chip/certification-tool/issues/888

### Testing
TC-TLSCERT-2.1 through 2.13 and TC-TLSCLIENT-3.x  now appears in the TH UI
  
Screenshot 2026-03-02 at 17 25 12

  Unit test added and passing
  
Screenshot 2026-03-02 at 17 22 15

### CODE REVIEW

File: test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py [hunk 1]
```diff
@@ -131,76 +132,128 @@ def load_include_list() -> set[str]:
...
+        ),
+        None,
+    )
+    if class_def is None:
+        return False
 
+    key = (id(module), class_name)
+    if key in _visiting:
         return False
+    _visiting = _visiting | {key}
```
[the remark this answers is not in the document]
Comment (rquidute) [reply test_collections/matter/sdk_tests/support/python_testing/list_python_tests_classes.py]: Done

File: app/tests/test_engine/test_list_python_tests_classes.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+        with patch(
+            "test_collections.matter.sdk_tests.support.python_testing"
+            ".list_python_tests_classes.PYTHON_TESTS_IGNORE_FILE",
+            f,
+        ):
+            result = load_ignore_list()
+        assert result == {"TC_FOO.py"}
+        assert "# this is a comment" not in result
+
+    def test_skips_blank_lines(self, tmp_path: Path) -> None:
```
[the remark this answers is not in the document]
Comment (rquidute) [reply app/tests/test_engine/test_list_python_tests_classes.py:99]: Done

File: app/tests/test_engine/test_python_test_initialization.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+                )
+
+                try:
+                    await manager.initialize_python_tests()
+                except Exception:
+                    # May fail due to import issues in test environment,
+                    # but we verify the attempt
+                    pass
+
+                # The flag should be set to True after initialization attempt
```
[the remark this answers is not in the document]
Comment (rquidute) [reply app/tests/test_engine/test_python_test_initialization.py]: Done

File: app/tests/test_engine/test_python_test_initialization.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+                "initialize_python_tests",
+                new_callable=AsyncMock,
+            ) as mock_init:
+                mock_init.return_value = (MagicMock(), MagicMock(), None)
+
+                try:
+                    await manager.initialize_python_tests()
+
+                    # Verify discover was called after initialization
+                    assert mock_discover.called, "Should call discover_test_collections"
```
[the remark this answers is not in the document]
Comment (rquidute) [reply app/tests/test_engine/test_python_test_initialization.py]: Done

**Review by oxesoft — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]
LGTM
